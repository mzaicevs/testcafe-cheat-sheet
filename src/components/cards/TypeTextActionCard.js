import React from "react";
import { actions_typeText } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";
import CardLink from "../../base_components/CardLink";

const TypeTextActionCard = () => (
  <CodeCard
    cardTitle={
      <CardLink
        text={actions_typeText.title}
        href={actions_typeText.fullDocLink}
      />
    }
    codeSnippet={{
      code: actions_typeText.code,
      language: actions_typeText.language,
      options: actions_typeText.options,
      example: actions_typeText.example
    }}
    description={actions_typeText.description}
  />
);

export default TypeTextActionCard;
