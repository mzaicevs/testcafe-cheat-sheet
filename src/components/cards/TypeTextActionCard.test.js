import React from "react";
import { shallow } from "enzyme";
import TypeTextActionCard from "./TypeTextActionCard";

describe("ClickActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<TypeTextActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
