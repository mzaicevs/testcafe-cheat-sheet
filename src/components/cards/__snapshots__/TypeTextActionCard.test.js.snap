// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle={
    <CardLink
      href="https://devexpress.github.io/testcafe/documentation/test-api/actions/type-text.html"
      text="Typing Action"
    />
  }
  codeSnippet={
    Object {
      "code": "t.typeText( selector, text [, options] )",
      "example": "import { Selector } from 'testcafe';

  const nameInput = Selector('#developer-name');
  
  fixture \`My fixture\`
      .page \`http://www.example.com/\`;
  
  test('Type and Replace', async t => {
      await t
          .typeText(nameInput, 'Tony')
          .typeText(nameInput, 'Stak', { replace: true })
          .typeText(nameInput, 'r', { caretPos: 3 })
          .expect(nameInput.value).eql('Stark');
  });",
      "language": "javascript",
      "options": "{
    modifiers: {
        ctrl: Boolean,
        alt: Boolean,
        shift: Boolean,
        meta: Boolean
    },

    offsetX: Number,
    offsetY: Number,
    caretPos: Number,
    replace: Boolean,
    paste: Boolean,
    speed: Number
}",
    }
  }
  description="Types the specified text into an input element."
/>
`;
